lms/djangoapps/instructor/views/api.py

```diff
@@ -2345,10 +2360,21 @@ def _list_instructor_tasks(request, course_id):
 
     Internal function with common code for both DRF and and tradition views.
     """
+    # This method is also used by other APIs with the GET method.
+    # The query_params attribute is utilized for GET requests,
+    # where parameters are passed as query strings.
+
     course_id = CourseKey.from_string(course_id)
     params = getattr(request, 'query_params', request.POST)
     problem_location_str = strip_if_string(params.get('problem_location_str', False))
     student = params.get('unique_student_identifier', None)
+
+    # For the DRF POST method, retrieve the data from request.data
+    if not student and not problem_location_str:
```

Comment:
Why not just pass the serializer from the `POST` method above?  You've already parsed this data out, may as well use that rather than re-getting it from the post data.

Reply:
Done, now picking validated-data.